src/pydvl/influence/base_influence_model.py

```diff
@@ -0,0 +1,173 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Collection, Generic, Iterable, Optional, TypeVar
+
+
+class InfluenceType(str, Enum):
+    r"""
+    Enum representation for the types of influence.
+
+    Attributes:
+        Up: Up-weighting a training point, see section 2.1 of
+            (Koh and Liang, 2017)<sup><a href="#koh_liang_2017">1</a></sup>
+        Perturbation: Perturb a training point, see section 2.2 of
+            (Koh and Liang, 2017)<sup><a href="#koh_liang_2017">1</a></sup>
+
+    """
+
+    Up = "up"
+    Perturbation = "perturbation"
+
+
+class UnsupportedInfluenceTypeException(ValueError):
+    def __init__(self, influence_type: str):
+        super().__init__(
+            f"Provided {influence_type=} is not supported. Choose one of InfluenceType.Up "
+            f"and InfluenceType.Perturbation"
+        )
+
+
+class NotFittedException(ValueError):
+    def __init__(self):
+        super().__init__(
+            f"Objects of type InfluenceFunctionModel must be fitted before calling influence methods. "
+            f"Call method fit with appropriate input."
+        )
+
+
+"""Type variable for tensors, i.e. sequences of numbers"""
+TensorType = TypeVar("TensorType", bound=Collection)
+DataLoaderType = TypeVar("DataLoaderType", bound=Iterable)
+
+
+class InfluenceFunctionModel(Generic[TensorType, DataLoaderType], ABC):
+    """
+    Generic abstract base class for computing influence related quantities. For a specific influence algorithm and
+    tensor framework, inherit from this base class
+    """
+
+    @property
+    @abstractmethod
+    def n_parameters(self):
+        """Number of trainable parameters of the underlying model"""
+
+    @property
+    @abstractmethod
+    def is_fitted(self):
+        """Overwrite this, to expose the fitting status of the instance."""
```

Comment:
I think you mean "override" (also: German comma ;) ). However, if an overriden method doesn't define a doscstring, the ide will show this one and that would be confusing. What about:
```suggestion
        """Whether or not the influence model has been fitted."""
```